uvtools/dspec.py

```diff
@@ -1237,15 +1686,490 @@ def delay_filter_leastsq(data, flags, sigma, nmax, add_noise=False, freq_units =
     for i in range(data.shape[0]):
         bf_model, cn_out, data_out = delay_filter_leastsq_1d(
             data[i], flags[i], sigma=sigma, nmax=(nmin, nmax), add_noise=add_noise, even_modes=even_modes,
-            use_linear=use_linear, cn_guess=cn_out, operator=F, freq_units=freq_units)
+            use_linear=use_linear, cn_guess=cn_out, operator=F, freq_units=freq_units, fundamental_period=fundamental_period)
         inp_data[i, :] = data_out
         cn_array[i, :] = cn_out
         mdl_array[i, :] = bf_model
 
     return mdl_array, cn_array, inp_data
 
 
-def delay_interpolation_matrix(nchan, ndelay, wgts, fundamental_period=None, cache={}, taper='none', return_diagnostics=False):
+def fit_basis_1d(x, y, w, filter_centers, filter_half_widths,
+                basis_options, suppression_factors=None,
+                method='leastsq', basis='dft', cache=None):
+    """
+    A 1d linear-least-squares fitting function for computing models and residuals for fitting of the form
+    y_model = A @ c
+    where A is a design matrix encoding our choice for a basis functions
+    and y_model
```

Comment:
unfinished sentence

Reply:
Thanks for catching this. I've finished it.